code/game/objects/items/hazardtape.dm

```diff
@@ -0,0 +1,258 @@
+
+		cur = start
+		var/tapetest = 0
+		while (cur!=end)
+			for(var/obj/item/holotape/Ptest in cur)
+				if(Ptest.icon_state == "[Ptest.icon_base]_[dir]")
+					tapetest = 1
+			if(tapetest != 1)
+				var/obj/item/holotape/P = new tape_type(cur)
+				P.icon_state = "[P.icon_base]_[dir]"
+			cur = get_step_towards(cur,end)
+
+		usr << "<span class='notice'>You finish project the legnth of [icon_base] holotape.</span>"
+		user.visible_message("<span class='warning'>[user] finishes projecting the length of [icon_base] holotape.</span>")
+
+		charging = 1
+		spawn(40)
+			charging = 0
+
+/obj/item/tapeproj/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
+	if(charging)
+		return
+
+	if(proximity_flag == 0) // not adjacent
+		return
+
+	if(istype(target, /obj/machinery/door/airlock) || istype(target, /obj/machinery/door/firedoor))
+		var/turf = get_turf(target)
+
+		if(locate(tape_type) in turf) //Don't you dare stack tape
+			return
+
+		user << "<span class='notice'>You start projecting the [icon_base] holotape onto [target].</span>"
+
+		if(!do_mob(user, target, 30))
+			return
+
+		var/atom/tape = new tape_type(turf)
+		tape.icon_state = "[icon_base]_door"
+		tape.layer = 3.2
+
+		user << "<span class='notice'>You project the [icon_base] holotape onto [target].</span>"
+
+		charging = 1
+		spawn(40)
+			charging = 0
+
+/obj/item/holotape/Bumped(var/mob/M)
+	if(src.allowed(M))
+		var/turf/T = get_turf(src)
+		M.loc = T
+
+/obj/item/holotape/CanPass(atom/movable/mover, turf/target, height=0, air_group=0)
+	if(!density) return 1
+	if(air_group || (height==0)) return 1
+
+	if((mover.flags & 2 || istype(mover, /obj/effect/meteor) || mover.throwing == 1) )
+		return 1
+	else
+		return 0
+
+/obj/item/holotape/attack_hand(mob/living/user)
+	if(user.a_intent == "help" && src.allowed(user) && src.density)
+		user.show_viewers("<span class='notice'>[user] lifts [src], allowing passage.</span>")
+		src.density = 0
+		spawn(200)
+			src.density = 1
+	else
+		user.changeNext_move(CLICK_CD_MELEE)
+		user.do_attack_animation(src)
+		playsound(loc, 'sound/weapons/Egloves.ogg', 80, 1)
+		user.visible_message("<span class='warning'>[user] hits [src].</span>", \
+							 "<span class='warning'>You hit [src].</span>" )
+
+		health -= rand(1,2)
+		healthcheck()
+
+/obj/item/holotape/proc/healthcheck()
+	if(health <= 0)
+		breaktape()
+
+/obj/item/holotape/ex_act(severity, target)
+	breaktape()
+
+/obj/item/holotape/blob_act()
+	breaktape()
+
+/obj/item/holotape/attack_paw(mob/living/user)
+	attack_hand(user)
+
+/obj/item/holotape/bullet_act(var/obj/item/projectile/Proj)
+	if(!Proj)
+		return
+	..()
+	if((Proj.damage_type != STAMINA))
+		src.health -= Proj.damage*0.3
+		healthcheck()
+	return
+
+/obj/item/holotape/attackby(obj/item/weapon/W as obj, mob/user as mob)
+	user.changeNext_move(CLICK_CD_MELEE)
+	add_fingerprint(user)
+	switch(W.damtype)
+		if(STAMINA)
+			return
+		if(BURN)
+			playsound(loc, 'sound/items/welder.ogg', 80, 1)
+		else
```

Comment:
Is the sound stuff necessary? Wouldn't it just play the hitsound of the attacking object anyway?